fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -1371,38 +1339,48 @@ private ScalarOperator addJoinDerivePredicate(
                 continue;
             }
 
-            List<ColumnRefOperator> leftJoinColumns = joinDeriveContext.getLeftJoinColumns();
-            List<ColumnRefOperator> rightJoinColumns = joinDeriveContext.getRightJoinColumns();
             Optional<ScalarOperator> derivedPredicateOpt = Optional.empty();
-            if (mvJoinOp.isLeftOuterJoin() && queryJoinOp.isInnerJoin()) {
-                // mv: left outer join , query: inner join
-                derivedPredicateOpt = getDerivedPredicate(rewriteContext,
-                        rewriter, mvColumnRefToScalarOp, rightJoinColumns, predicates, true, true, false);
-            } else if (mvJoinOp.isLeftOuterJoin() && queryJoinOp.isLeftAntiJoin()) {
-                // mv: left outer join , query: left anti join
-                derivedPredicateOpt = getDerivedPredicate(rewriteContext,
-                        rewriter, mvColumnRefToScalarOp, rightJoinColumns, predicates, false, true, false);
-            } else if (mvJoinOp.isRightOuterJoin() && queryJoinOp.isInnerJoin()) {
-                // mv: right outer join , query: inner join
-                derivedPredicateOpt = getDerivedPredicate(rewriteContext,
-                        rewriter, mvColumnRefToScalarOp, leftJoinColumns, predicates, true, true, false);
-            } else if (mvJoinOp.isRightOuterJoin() && queryJoinOp.isRightAntiJoin()) {
-                // mv: right outer join , query: right anti join
-                derivedPredicateOpt = getDerivedPredicate(rewriteContext,
-                        rewriter, mvColumnRefToScalarOp, leftJoinColumns, predicates, false, true, false);
-            } else if (mvJoinOp.isFullOuterJoin() && queryJoinOp.isLeftOuterJoin()) {
-                // mv: full outer join , query: left outer join
-                derivedPredicateOpt = getDerivedPredicate(rewriteContext,
-                        rewriter, mvColumnRefToScalarOp, leftJoinColumns, predicates, true, false, true);
-            } else if (mvJoinOp.isFullOuterJoin() && queryJoinOp.isRightOuterJoin()) {
-                // mv: full outer join , query: right outer join
-                derivedPredicateOpt = getDerivedPredicate(rewriteContext,
-                        rewriter, mvColumnRefToScalarOp, rightJoinColumns, predicates, true, false, true);
-            } else if (mvJoinOp.isFullOuterJoin() && queryJoinOp.isInnerJoin()) {
-                // mv: full outer join , query: inner join
-                // add right table's not null constrains
-                derivedPredicateOpt = getDerivedPredicate(rewriteContext,
-                        rewriter, mvColumnRefToScalarOp, rightJoinColumns, predicates, true, false, true);
+            if (joinDeriveContext.getMvJoinType().isLeftOuterJoin() && joinDeriveContext.getQueryJoinType().isInnerJoin()) {
```

Comment:
Save `joinDeriveContext.getMvJoinType()` and `joinDeriveContext.getQueryJoinType()` as variables to avoid calling multi times?